lib/request-handler.js

```diff
@@ -16,20 +16,22 @@ var retryOnNextHostDecision = {
 
 /**
  * Handles a request to Cassandra, dealing with host fail-over and retries on error
- * @param {?Client} client Client instance used to retrieve and set the keyspace. It can be null.
+ * @param {Client} client Client instance used to retrieve and set the keyspace.
  * @param {Object} options
  * @constructor
  */
 function RequestHandler(client, options) {
   this.client = client;
-  this.loadBalancingPolicy = options.policies.loadBalancing;
-  this.retryPolicy = options.policies.retry;
+  this.profileManager = client.profileManager;
+  var profile = client.profileManager.getDefault();
+  this.loadBalancingPolicy = profile.loadBalancing;
+  this.retryPolicy = profile.retry;
```

Comment:
Shouldn't retryPolicy be derived from options.retry instead?  Otherwise query level retry will not be used, from client-options#createQueryOptions#160:

```javascript
retry: ifUndefined3(userOptions.retry, profile.retry, defaultQueryOptions.retry),
```